index.ts

```diff
@@ -0,0 +1,552 @@
+  sound?: boolean; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+  // single-target
+  'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+  // single-target, automatic
+  'self'|'randomNormal'|
+  // spread
+  'allAdjacent'|'allAdjacentFoes'|
+  // side and field
+  'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+  volatileStatus?: ID;
+  boosts?: Partial<BoostsTable>;
+}
+
+interface SecondaryEffect {
+  chance?: number;
+  boosts?: Partial<BoostsTable>;
+  self?: SelfEffect;
+  status?: StatusName;
+  volatileStatus?: ID;
+}
+
+type ContestType = 'Cool' | 'Beautiful' | 'Cute' | 'Clever' | 'Tough';
+
+interface CommonMove<TypeT, ItemT, MoveT> {
+  accuracy: number | true;
+  basePower: number;
+  flags: MoveFlags;
+  pp: number;
+  target: MoveTarget;
+  type: TypeT;
+
+  alwaysCrit?: boolean;
+  alwaysHit?: boolean;
+  basePowerModifier?: number;
+  boosts?: Partial<BoostsTable> | false
+  bypassesProtect?: boolean;
+  category: MoveCategory;
+  contestType?: ContestType;
+  critModifier?: number;
+  critRatio?: number;
+  damage?: number | 'level';
+  defensiveCategory?: MoveCategory;
+  drain?: [number, number]; // fraction
+  forceSTAB?: boolean;
+  forceSwitch?: boolean;
+  givesHealth?: boolean;
+  hasCustomRecoil?: boolean;
+  hasPriority?: boolean;
+  hasSecondaryEffect?: boolean;
+  heal?: [number, number]; // fraction
+  ignoreAbility?: boolean;
+  ignoreAccuracy?: boolean;
+  ignoreDefensive?: boolean;
+  ignoreEvasion?: boolean;
+  ignoreImmunity?: boolean | TypeT;
+  ignoreNegativeOffensive?: boolean;
+  ignoreOffensive?: boolean;
+  ignorePositiveDefensive?: boolean;
+  ignorePositiveEvasion?: boolean;
+  ignoresBurn?: boolean;
+  ignoresDefenseBoosts?: boolean;
+  infiltrates?: boolean;
+  isFutureMove?: boolean;
+  isSpread?: boolean | 'allAdjacent';
+  isZ?: ItemT;
+  multiaccuracy?: boolean;
+  multihit?: number | [number, number]; // range
+  noCopy?: boolean;
+  noDamageVariance?: boolean;
+  noFaint?: boolean;
+  noMetronome?: MoveT[];
+  noPPBoosts?: boolean;
+  noSketch?: boolean;
+  nonGhostTarget?: string;
+  ohko?: true | 'Ice'; // TODO: better type here?
+  percentHealed?: number;
+  pressureTarget?: string;
+  priority?: number;
+  pseudoWeather?: string:
+  recoil?: [number, number]; // fraction
+  secondaries?: SecondaryEffect[];
+  self?: SelfEffect;
+  selfBoost?: {boosts?: Partial<BoostsTable>};
+  selfSwitch?: ID | boolean;
+  selfdestruct?: 'always' | 'ifHit';
+  sideCondition?: ID;
+  sleepUsable?: boolean;
+  slotCondition?: string;
+  spreadHit: boolean;
+  spreadModifier?: number;
+  stallingMove?: boolean;
+  status?: StatusName;
+  stealsBoosts?: boolean;
+  terrain?: string;
+  thawsTarget?: boolean;
+  useSourceDefensive?: boolean;
+  useTargetOffensive?: boolean;
+  usesHighestAttackStat?: boolean;
+  volatileStatus?: ID;
+  weather?: WeatherName;
+  willCrit?: boolean;
```

Comment:
moveflag

Reply:
this seems to be a similar case to the 'bypasses-accuracy' situation